proposed/2023/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,199 @@
+
+
+#### Group Ownership
+There are few interesting scenarios related to package ownership:
+1. The user has staged a package to a group for later publishing, but the ownership has changed, and the user is no longer
+ an owner for a package in a group. At this point, publishing request should fail with a message that the group has been modified
+ from its original set, and the outcome of the request may be different from what the user was expecting, indicating the package
+ that was moved out of the group. If the user is aware of the problem and still wants to move forward with the change,
+ they will have to pass an additional `--force` parameter to the publish command: `nuget publish <group-id> --force`.
+2. The group to be published no longer has any packages in it. In this case the command should fail detailing the reason for the failure.
+3. This will be the same but for deprecation. Essentially, any bulk operation using groups, where the set of the packages in the group
+ have changed after staging, should result in an error with details about the change. The `--force` flag should again be used
+ to ignore the change and move forward with execution of the requst (either publishing or deprecation). To summarize,
+ the fundamental principle is that a user cannot deprecate a package they don't own.
+
+#### Lifespan of a staged group
+If users keep staging packages but do nothing with them, NuGet will become an ever-growing package graveyard,
+ incurring higher and higher costs over time simply for storing these packages. To avoid this problem, staging should be time-limited.
+ That is, there should be a maximum lifespan for a staged package, that is not published. NuGet should then unstage / remove
+ any package that was staged but never published within the specified / pre-configured time period. As a maximum timeslot
+ a default one month period can be considered. However, to reduce the cost of maintaining these packages,
+ the `nuget stage` command can also accept an optional `--lifespan [number of days]` parameter,
+ which can indicate a shorter time period, thus giving NuGet ability to remove those packages earlier than the maximum period.
+
+Note, that this opens up a new scenario, where a group has been created with a package 29 days ago, and then a new package is staged to the same group.
+In two days from that time, the package author tries to publish the staged group. At this point, the publish command will fail,
+indicating that the group has been modified, and showing the package that has been deleted by NuGet out of the staging due to the expiration.
+The user can still continue publishing using the `--force` option, which will result in the non-expired/ removed staged packages to be published.
+Alternatively, the user can restage the expired package and call the publish command again.
+
+#### Expanding groups
```

Comment:
Not sure if you answered these in the spec already, but some questions:
1. Can one put multiple versions of the same package ID into the group?
2. Will normal push (publishing as it exists today without staging) of same version allowed, if that version exists in the staged group?
3. What happens if a later version is published directly than what a staged group has? Will publishing of a newer version impact the staged group in anyway?
4. Can one member of org stage the group of packages and another member of the org publish? Will this require `--force` option?
5. What happens to the package ID namespace if a staged group is discarded? Same question for package version? Are they released for all?